modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -74,32 +109,53 @@ def set_module_options(mod)
   end
 
   def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
+    return false unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    is_session_type_compat?(mod) &&
+      (@validate_platform ? is_module_platform?(mod) : true) &&
+      (@validate_arch ? is_module_arch?(mod) : true) &&
       is_module_options_ready?(mod)
-    )
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    setup_validation_options
+    setup_color_options
+
     # Collects exploits into an array
     @local_exploits = []
-    framework.exploits.each do |name, _obj|
+    framework.exploits.each_with_index do |(name, _obj), index|
```

Comment:
For visibility here's `filter_map` in Ruby 2.7+

https://blog.saeloun.com/2019/05/25/ruby-2-7-enumerable-filter-map.html

We still support Ruby 2.6 though, so there's nothing to action over here 😄